src/lib/trace/config.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/spf13/viper"
+)
+
+const (
+	TraceEnvPrefix = "trace"
+)
+
+// C is the global configuration for trace
+var C Config
+
+func init() {
+	viper.SetConfigType("json")
+	viper.SetEnvPrefix(TraceEnvPrefix)
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	viper.AutomaticEnv()
+	C = Config{Otel: OtelConfig{}, Jaeger: JaegerConfig{}}
+	C.Enabled = viper.GetBool("enabled")
+	C.SampleRate = viper.GetFloat64("sample_rate")
+	C.Namespace = viper.GetString("namespace")
+	C.ServiceName = viper.GetString("service_name")
+	C.Jaeger.Endpoint = viper.GetString("jaeger_endpoint")
+	C.Jaeger.Username = viper.GetString("jaeger_agent_username")
+	C.Jaeger.Password = viper.GetString("jaeger_agent_password")
+	C.Jaeger.AgentHost = viper.GetString("jaeger_agent_host")
+	C.Jaeger.AgentPort = viper.GetString("jaeger_agent_port")
+	C.Otel.Endpoint = viper.GetString("otel_endpoint")
+	C.Otel.URLPath = viper.GetString("otel_url_path")
+	C.Otel.Compression = viper.GetBool("otel_compression")
+	C.Otel.Insecure = viper.GetBool("otel_insecure")
+	C.Otel.Timeout = viper.GetInt("otel_timeout")
+	var jsonExample = []byte(viper.GetString("attributes"))
+	viper.ReadConfig(bytes.NewBuffer(jsonExample))
+	fmt.Println(viper.GetStringMapString("attributes"))
+	C.Attributes = viper.GetStringMapString("attributes")
+	log.Infof("ns: %s attr %+v", C.Namespace, C.Attributes)
```

Comment:
All these configure items should be put into the system config.
Refer https://github.com/goharbor/harbor/blob/107e468b6017c74ca899ef90414789d7a0d90b01/src/lib/config/metadata/metadatalist.go#L66